File: BE/routine-service/src/main/java/com/hogumiwarts/lumos/routine/dto/RoutineDetailResponse.java
package com.hogumiwarts.lumos.routine.dto;

import java.util.List;

import com.hogumiwarts.lumos.routine.entity.Routine;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Schema(description = "루틴 응답 DTO")
public class RoutineDetailResponse {

	@Schema(description = "루틴 이름", example = "수면 루틴")
	private String routineName;

	@Schema(description = "루틴 아이콘", example = "lamp")
	private String routineIcon;

	@Schema(description = "기기 정보")
	private List<DevicesCreateResponse> devices;

	@Schema(description = "제스처 ID", example = "1")
	private Long gestureId;

	@Schema(description = "제스처 이름", example = "핑거스냅")
	private String gestureName;

	@Schema(description = "제스처 이미지 URL", example = "https://example.com/img/gesture_snap.png")
	private String gestureImageUrl;

	@Schema(description = "제스처 설명", example = "손가락을 튕깁니다.")
	private String gestureDescription;

	public static RoutineDetailResponse from(Routine routine, GestureResponse gestureResponse, List<DevicesCreateResponse> devices) {
		return RoutineDetailResponse.builder()
			.routineName(routine.getRoutineName())
			.routineIcon(routine.getRoutineIcon())
			.devices(devices)
			.gestureId(routine.getGestureId())
			.gestureName(gestureResponse != null ? gestureResponse.getGestureName() : null)
			.gestureImageUrl(gestureResponse != null ? gestureResponse.getGestureImageUrl() : null)
			.gestureDescription(gestureResponse != null ? gestureResponse.getGestureDescription() : null)
			.build();
	}
}

File: Android/data/src/main/java/com/hogumiwarts/data/entity/remote/Response/routine/RoutineDetailResponse.kt
package com.hogumiwarts.data.entity.remote.Response.routine

import com.hogumiwarts.domain.model.routine.CommandData

data class RoutineDetailResponse(
    val status: Int,
    val message: String,
    val data: RoutineDetailData
)

data class RoutineDetailData(
    val routineName: String,
    val routineIcon: String,
    val devices: List<RoutineDeviceData>,
    val gestureId: Int,
    val gestureName: String,
    val gestureImageUrl: String,
    val gestureDescription: String
)

data class RoutineDeviceData(
    val deviceId: Long,
    val deviceName: String,
    val deviceType: String,
    val deviceImageUrl: String?,
    val commands: List<CommandData>
)
